client/src/nv_ingest_client/primitives/jobs/job_spec.py

```diff
@@ -5,81 +5,91 @@
 from uuid import UUID
 
+from nv_ingest_client.primitives.tasks.caption import CaptionTask
+from nv_ingest_client.primitives.tasks.dedup import DedupTask
+from nv_ingest_client.primitives.tasks.embed import EmbedTask
+from nv_ingest_client.primitives.tasks.filter import FilterTask
+from nv_ingest_client.primitives.tasks.split import SplitTask
+from nv_ingest_client.primitives.tasks.store import StoreTask
+from nv_ingest_client.primitives.tasks.vdb_upload import VdbUploadTask
+from nv_ingest_client.primitives.tasks.extract import ExtractTask
 from nv_ingest_client.primitives.tasks import Task
+from pydantic import BaseModel
 
 logger = logging.getLogger(__name__)
 
 
-class JobSpec:
+class JobSpec(BaseModel):
     """
     Specification for creating a job for submission to the nv-ingest microservice.
 
-    Parameters
-    ----------
-    payload : Dict
-        The payload data for the job.
-    tasks : Optional[List], optional
-        A list of tasks to be added to the job, by default None.
-    source_id : Optional[str], optional
-        An identifier for the source of the job, by default None.
-    job_id : Optional[UUID], optional
-        A unique identifier for the job, by default a new UUID is generated.
-    extended_options : Optional[Dict], optional
-        Additional options for job processing, by default None.
-
     Attributes
     ----------
-    _payload : Dict
-        Storage for the payload data.
-    _tasks : List
-        Storage for the list of tasks.
-    _source_id : str
-        Storage for the source identifier.
-    _job_id : UUID
-        Storage for the job's unique identifier.
-    _extended_options : Dict
+    document_type: str
+        Type of document that is being submitted.
+    extended_options : Dict
         Storage for the additional options.
+    job_id : UUID
+        Storage for the job's unique identifier.
+    payload : str
+         Storage for the payload data.
+    source_id : str
+         Storage for the source identifier.
+    source_name: str
+        Storage for the source name.
+    tasks : List
+        Storage for the list of tasks.
 
     Methods
     -------
-    to_dict() -> Dict:
-        Converts the job specification to a dictionary.
     add_task(task):
         Adds a task to the job specification.
     """
 
-    def __init__(
-        self,
-        payload: str = None,
-        tasks: Optional[List] = None,
-        source_id: Optional[str] = None,
-        source_name: Optional[str] = None,
-        document_type: Optional[str] = None,
-        job_id: Optional[typing.Union[UUID, str]] = None,
-        extended_options: Optional[Dict] = None,
-    ) -> None:
-        self._document_type = document_type or "txt"
-        self._extended_options = extended_options or {}
-        self._job_id = job_id
-        self._payload = payload
-        self._source_id = source_id
-        self._source_name = source_name
-        self._tasks = tasks or []
+    document_type: Optional[str]
```

Comment:
I am not a Pydantic expert but I tried so so so many things to avoid changing this object at all. However, without these changes, mainly removing the `_attribute` -> `attribute` I was unable to get FastAPI to accept the objects.

If someone can make that work without those changes I would certainly appreciate that assist here.